recipes/_compute_slurm_config.rb

```diff
@@ -31,7 +31,16 @@
   only_if { ::File.exist?('/bin/systemctl') }
 end
 
-service "slurm" do
-  supports restart: false
-  action %i[enable]
+if ::File.exist?('/bin/systemctl') 
```

Comment:
Isn't there a better test for init style?

Reply:
https://github.com/awslabs/cfncluster-cookbook/blob/develop/recipes/_ganglia_install.rb#L59

https://docs.chef.io/ohai.html using the init_package.rb plugin.

Reply:
yeah I saw that. but https://github.com/awslabs/cfncluster-cookbook/blob/develop/recipes/_ganglia_install.rb#L65 https://github.com/awslabs/cfncluster-cookbook/blob/develop/recipes/_ganglia_install.rb#L71 and 
https://github.com/awslabs/cfncluster-cookbook/blob/develop/recipes/_ganglia_install.rb#L78
made me unsure if this was reliable.
I googled a bit about this issue and became convinced checking for systemctl was the most reliable way to determine this, but I am not clear on ohai, will have a look.

Reply:
I think you're seeing two different things.  The ohai check is the right way to check for init system type from Chef.  The things on line 65 and 71 you reference are to not overwrite an existing startup file.  Remember that these recipes work in many contexts, and the starting state isn't always known, so some defensive programming may be the right thing.